catalog/core/catalog-core-contentresourcereader/src/main/java/org/codice/ddf/catalog/content/resource/reader/ContentResourceReader.java

```diff
@@ -111,10 +115,26 @@ public String getVersion() {
 
     @Override
     public Set<String> getOptions(Metacard metacard) {
-        LOGGER.trace("ENTERING/EXITING: getOptions");
-        LOGGER.debug(
-                "ContentResourceReader getOptions doesn't support options, returning empty set.");
-
+        if (metacard != null
+                && metacard.getAttribute(Metacard.DERIVED_RESOURCE_DOWNLOAD_URL) != null
+                && !CollectionUtils.isEmpty(metacard.getAttribute(Metacard.DERIVED_RESOURCE_DOWNLOAD_URL)
+                .getValues())) {
+            Set<String> options = new HashSet<>();
+            for (Serializable value : metacard.getAttribute(Metacard.DERIVED_RESOURCE_DOWNLOAD_URL)
+                    .getValues()) {
+                try {
+                    URI contentUri = new URI((String) value);
+                    if (ContentItem.CONTENT_SCHEME.equals(contentUri.getScheme())) {
+                        if (StringUtils.isNotBlank(contentUri.getFragment())) {
+                            options.add(contentUri.getFragment());
+                        }
+                    }
+                } catch (URISyntaxException e) {
+                    // Ignore - nothing to do
```

Comment:
I wonder if we should still log something just in case someone accidentally overwrite the wrong field with a non-URI value.